greedy_methods/minimum_waiting_time.py

```diff
@@ -0,0 +1,57 @@
+"""
+This is a pure Python implementation for minimum waiting time problem using greedy
+algorithm.
+reference: https://www.youtube.com/watch?v=Sf3eiO12eJs
+
+For doctests run following command:
+python -m doctest -v minimum_waiting_time.py
+
+The minimum_waiting_time function uses a greedy algorithm to calculate the minimum
+time for queries to complete. It sorts the list in non-decreasing order, calculates
+the waiting time for each query by multiplying its position in the list with the
+sum of all remaining query times, and returns the total waiting time. A doctest
+ensures that the function produces the correct output.
+"""
+
+
+def minimum_waiting_time(queries: list[int]) -> int:
+    """
+    This function takes a list of query times and returns the minimum waiting time
+    for all queries to be completed.
+
+    Args:
+        queries [list[int]]: A list of queries
+
+    Returns:
+        total_waiting_time [int]: Minimum waiting time
```

Comment:
Please do not repeat the data types that are already stated in the function signature because users do not know which one to trust when one changes but not the other.  Please replace the `picoseconds`.
```suggestion
        queries: A list of queries measured in picoseconds

    Returns:
        total_waiting_time: Minimum waiting time measured in picoseconds
```